CSharpMath.Editor.Tests/CaretTests.cs

```diff
@@ -176,4 +175,103 @@ public class CaretCanStartAndStopBlinking {
+      var outer = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(keyboard.MathList));
+      var inner = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(outer.Subscript));
+      Assert.Equal(MathKeyboardCaretState.ShownThroughPlaceholder, keyboard.CaretState);
+      Assert.Equal("😐", outer.Nucleus);
+      Assert.Equal(System.Drawing.Color.Blue, outer.Color);
+      Assert.Equal("😀", inner.Nucleus);
+      Assert.Equal(System.Drawing.Color.Green, inner.Color);
+
+      await Task.Delay((int)MathKeyboard<TestFont, char>.DefaultBlinkMilliseconds + CaretBlinks.MillisecondBuffer);
+      Assert.Equal(MathKeyboardCaretState.TemporarilyHidden, keyboard.CaretState);
+      Assert.Equal("😐", outer.Nucleus);
+      Assert.Equal(System.Drawing.Color.Blue, outer.Color);
+      Assert.Equal("😐", inner.Nucleus);
+      Assert.Equal(System.Drawing.Color.Blue, inner.Color);
+
+      await Task.Delay((int)MathKeyboard<TestFont, char>.DefaultBlinkMilliseconds + CaretBlinks.MillisecondBuffer);
+      Assert.Equal(MathKeyboardCaretState.ShownThroughPlaceholder, keyboard.CaretState);
+      Assert.Equal("😐", outer.Nucleus);
+      Assert.Equal(System.Drawing.Color.Blue, outer.Color);
+      Assert.Equal("😀", inner.Nucleus);
+      Assert.Equal(System.Drawing.Color.Green, inner.Color);
+
+      RestorePlaceholderDefaultSettings();
+    }
+    [Fact]
+    public void AllCustomizablePlaceholderPropertiesAreResetOnCaretVisible() {
+      LaTeXSettings.PlaceholderActiveNucleus = "😀";
+      LaTeXSettings.PlaceholderRestingNucleus = "😐";
+      LaTeXSettings.PlaceholderActiveColor = System.Drawing.Color.Green;
+      LaTeXSettings.PlaceholderRestingColor = System.Drawing.Color.Blue;
+      var keyboard = new MathKeyboard<TestFont, char>(TestTypesettingContexts.Instance, new TestFont()) {
+        CaretState = MathKeyboardCaretState.Shown
+      };
+      Assert.Equal(MathKeyboardCaretState.Shown, keyboard.CaretState);
+      keyboard.KeyPress(MathKeyboardInput.Subscript);
+      var outer = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(keyboard.MathList));
+      var inner = Assert.IsType<Atom.Atoms.Placeholder>(Assert.Single(outer.Subscript));
+      Assert.Equal(MathKeyboardCaretState.ShownThroughPlaceholder, keyboard.CaretState);
+
+      keyboard.InsertionIndex = MathListIndex.Level0Index(keyboard.MathList.Count);
+      Assert.Equal(MathKeyboardCaretState.Shown, keyboard.CaretState);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, outer.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, outer.Color);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingNucleus, inner.Nucleus);
+      Assert.Equal(LaTeXSettings.PlaceholderRestingColor, inner.Color);
+      RestorePlaceholderDefaultSettings();
+    }
+    [Fact]
+    public void LaTeXSettings_Placeholder_IsNewInstance() {
+      Assert.False(LaTeXSettings.Placeholder == LaTeXSettings.Placeholder);
+      // Double check, also verify that its contents are 'fresh':
```

Comment:
test name doesn't correspond the code below

Reply:
nope, sorry - my mistake, wasn't look carefully

Reply:
Though I still dislike this part: no guarantee that `Assert.NotEqual("x", LaTeXSettings.Placeholder.Nucleus);`, etc is true.

Reply:
Agreed. I will replace the Assert.NotEqual in my next commit.